cpp/include/cudf/io/csv.hpp

```diff
@@ -1758,6 +1758,14 @@ class csv_writer_options_builder {
 void write_csv(csv_writer_options const& options,
                rmm::cuda_stream_view stream = cudf::get_default_stream());
 
+/**
+ * @brief Checks if a cudf::data_type is supported for CSV writing.
+ *
+ * @param type The data_type to check.
+ * @return true if the type is supported for CSV writing, false otherwise.
+ */
+bool is_csv_writable_type(data_type type);
```

Comment:
Oh, man. I recently added APIs with similar semantics, but completely different naming https://github.com/rapidsai/cudf/pull/18578
Should we try to make the names consistent? i.e. this would become `is_supported_write_csv(data_type type)`. IMO it should be fine to rely on overloads and exclude what is being checked in the function name, but I'm very open to discussion.

Reply:
Lol yeah we can. I do like `is_csv_writable_type` better though

Reply:
Done

Reply:
> I do like is_csv_writable_type better though

It is better worded, just maybe harder to keep consistent.